myriad-scheduler/src/main/java/org/apache/myriad/scheduler/fgs/YarnNodeCapacityManager.java

```diff
@@ -98,6 +105,45 @@ public boolean allowCallBacksForNode(NodeId nodeManager) {
     };
   }
 
+  private Protos.TaskID containerToTaskId(RMContainer container) {
+    return Protos.TaskID.newBuilder().setValue("yarn_" + container.getContainerId()).build();
+  }
+
+  @Override
+  public void beforeReleaseContainers(List<ContainerId> containerIds, SchedulerApplicationAttempt attempt) {
+    //NOOP beforeCompletedContainer does this
+  }
+
+  @Override
+  public void beforeCompletedContainer(RMContainer rmContainer, ContainerStatus containerStatus, RMContainerEventType type) {
+    if (type.equals(RMContainerEventType.KILL) || type.equals(RMContainerEventType.RELEASED)) {
+      LOGGER.info("{} completed with exit status {}, killing cooresponding mesos task.", rmContainer.getContainerId().toString(), type);
+      myKillContainer(rmContainer);
+    }
+  }
+
+  private synchronized void myKillContainer(RMContainer rmContainer) {
+    if (rmContainer != null && rmContainer.getContainer() != null) {
+      Protos.TaskID taskId = containerToTaskId(rmContainer);
+      //TODO (darinj) Reliable messaging
```

Comment:
It's slightly unreliable in the event of a network partition, but it doesn't matter now as the zero-size-nodemanger immediately goes to TASK_LOST.  I'm looking at the StateStore to see about adding a timeout delay for node managers for brief partitions, at that point we'll need to have a thread monitor a kill queue, while a TASK_KILLED event would remove the task from the kill queue.